src/Core/src/Platform/iOS/MauiView.cs

```diff
@@ -7,106 +7,258 @@
+		/// Used to avoid redundant measure calls when constraints haven't changed.
+		/// NaN indicates no previous measure has been performed.
+		/// </summary>
 		double _lastMeasureWidth = double.NaN;
 
+		/// <summary>
+		/// Current safe area padding values (top, left, bottom, right) in device-independent units.
+		/// These values represent the insets needed to avoid system UI elements like status bars,
+		/// navigation bars, and home indicators.
+		/// </summary>
+		SafeAreaPadding _safeArea = SafeAreaPadding.Empty;
+
+		/// <summary>
+		/// Flag indicating that the safe area has changed and needs to be revalidated.
+		/// Set to true when SafeAreaInsetsDidChange() is called by the system.
+		/// </summary>
+		bool _safeAreaInvalidated = true;
+
+		/// <summary>
+		/// Flag indicating whether this view should apply safe area adjustments to its layout.
+		/// This is true when the view implements ISafeAreaView, doesn't ignore safe area,
+		/// and the safe area is not empty.
+		/// </summary>
+		bool _appliesSafeAreaAdjustments;
+
+
+		/// <summary>
+		/// Weak reference to the cross-platform IView that this native view represents.
+		/// Uses WeakReference to avoid circular references between platform and cross-platform layers.
+		/// </summary>
 		WeakReference<IView>? _reference;
+
+		/// <summary>
+		/// Weak reference to the cross-platform layout manager for this view.
+		/// Used to delegate measure and arrange operations to the cross-platform layout system.
+		/// </summary>
 		WeakReference<ICrossPlatformLayout>? _crossPlatformLayoutReference;
 
+		/// <summary>
+		/// Gets or sets the cross-platform IView that this native view represents.
+		/// This provides access to the MAUI view properties and behavior from the platform layer.
+		/// </summary>
 		public IView? View
 		{
 			get => _reference != null && _reference.TryGetTarget(out var v) ? v : null;
 			set => _reference = value == null ? null : new(value);
 		}
 
+		/// <summary>
+		/// Determines if this view has fixed constraints that prevent it from changing size.
+		/// Views with fixed constraints don't need to propagate measure invalidations to ancestors.
+		/// </summary>
 		bool HasFixedConstraints => CrossPlatformLayout is IConstrainedView { HasFixedConstraints: true };
 
+		/// <summary>
+		/// Determines if this view should respond to safe area changes.
+		/// Returns true if the view implements ISafeAreaView, doesn't ignore safe area,
+		/// and the current iOS version supports safe area insets.
+		/// </summary>
 		bool RespondsToSafeArea()
 		{
+			if (View is not ISafeAreaView sav || sav.IgnoreSafeArea)
+			{
+				return false;
+			}
+
 			if (_respondsToSafeArea.HasValue)
+			{
 				return _respondsToSafeArea.Value;
-			return (bool)(_respondsToSafeArea = RespondsToSelector(new Selector("safeAreaInsets")));
+			}
+
+			return (_respondsToSafeArea = RespondsToSelector(new Selector("safeAreaInsets"))).Value;
```

Comment:
From the top, is this needed? Should it also go in the ScrollView or should we delete? This is iOS 11+ and MacCat 13.1+.